### PR TITLE
Add getServiceNamesAndTypes method to Node class

### PR DESCRIPTION
**Pull request by ivanpauno**
Depends on https://github.com/osrf/ros2_java/pull/35.

### CODE REVIEW

File: rcljava/src/main/cpp/org_ros2_rcljava_node_NodeImpl.cpp [hunk 1]
```diff
@@ -410,3 +410,35 @@ Java_org_ros2_rcljava_node_NodeImpl_nativeGetSubscriptionNamesAndTypesByNode(
...
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_names_and_types_t service_names_and_types = rcl_get_zero_initialized_names_and_types();
+
+  rcl_ret_t ret = rcl_get_service_names_and_types_by_node(
+    node,
+    &allocator,
+    name,
+    namespace_,
+    &service_names_and_types);
+  RCLJAVA_COMMON_THROW_FROM_RCL_X(env, ret, "failed to get service names and types", goto cleanup);
```
Comment (jacobperron) [inline rcljava/src/main/cpp/org_ros2_rcljava_node_NodeImpl.cpp]: Consider using scoped exit from rcpputils.

**Review by jacobperron — APPROVED** [review]

**Comment by ivanpauno** [comment]
The PRs that are still a draft have to wait a bit for being reviewed (I need to merge some of the opened PRs in order to avoid having to solve ton of conflicts, I'm marking PRs as "ready for review" accordingly).
https://github.com/osrf/ros2_java/pull/27 is ready for being reviewed.

File: rcljava/src/main/cpp/org_ros2_rcljava_node_NodeImpl.cpp (hunk 1 above)
Comment (ivanpauno) [reply rcljava/src/main/cpp/org_ros2_rcljava_node_NodeImpl.cpp]: done in https://github.com/osrf/ros2_java/pull/36/commits/6e804e834fc5817d868c35123a6ce90b01fab3c9.

**Comment by ivanpauno** [comment]
Conflicts were resolved, this is ready for review now

**Review by jacobperron — APPROVED** [review]

**Pull request merged**